zentral/contrib/osquery/serializers.py

```diff
@@ -220,3 +221,45 @@ def update(self, instance, validated_data):
         if compliance_check_deleted:
             query.refresh_from_db()
         return query
+
+
+class PackQuerySerializer(serializers.ModelSerializer):
+    pack = serializers.PrimaryKeyRelatedField(queryset=Pack.objects.all())
+    query = serializers.PrimaryKeyRelatedField(queryset=Query.objects.all())
+
+    class Meta:
+        model = PackQuery
+        fields = "__all__"
+
+    def validate_query(self, value):
+        slug = slugify(value.name)
+        pq_qs = PackQuery.objects.all()
+        if self.instance:
+            pq_qs = pq_qs.exclude(id=self.instance.id)
+        if pq_qs.filter(slug=slug).exists():
+            raise serializers.ValidationError(["PackQuery with this Query already exists."])
```

Comment:
The problem is not that a pack query with the same query exists. The problem is that 2 different queries with names resulting in the same slug cannot be added to the same pack. This is a corner case that we need to solve properly. I have tested this, and the bug exists in the UI as well. A better solution would be to add the PK of the query to the slug when there is a conflict:
- If no conflict, slug
- If conflict, slug + PK of the query (maybe with a separator, like "-")

Reply:
ok, so basically to allow this by appending the pk to the slug name separated by "-" ..

Reply:
Yes, allow this, only append the PK if there is a conflict.